blocks/shopify/includes/options/store/class-store.php

```diff
@@ -0,0 +1,124 @@
+<?php
+/**
+ * Store
+ *
+ * @package Cata\Blocks\Shopify\Options
+ * @since 0.8.2
+ */
+
+namespace Cata\Blocks\Shopify\Options;
+
+use Cata\Blocks\Shopify\Options\Add_Store;
+
+/**
+ * Store
+ */
+class Store {
+
+	/**
+	 * Setting Name
+	 * Unique key for a single option/setting, stored in DB.
+	 */
+	const SETTING_NAME = 'cata_blocks_shopify_stores';
+
+	/**
+	 * Register
+	 * 
+	 * @param string $option_group
+	 */
+	public static function register( string $option_group ) : void {
+		register_setting(
+			$option_group,
+			self::SETTING_NAME,
+			array(
+				'type'              => 'string',
+				'sanitize_callback' => array( __CLASS__, 'sanitize_settings' ),
+				'show_in_rest'      => true,
+				'default'           => '',
+			)
+		);
+
+		add_settings_field(
+			self::SETTING_NAME,
+			'Shopify Stores',
+			array( __CLASS__, 'field_callback' ),
+			$option_group,
+			$option_group . '_section',
+			array(
+				'label_for' => self::SETTING_NAME,
+				'id'        => self::SETTING_NAME,
+				'name'      => self::SETTING_NAME,
+			)
+		);
+	}
+
+	/**
+	 * Sanitize Settings
+	 * 
+	 * @param array $array
+	 * @return array
+	 */
+	public static function sanitize_settings( $array ) : string {
+		if ( empty( $array ) ) {
+			return array();
+		}
+
+		$array = map_deep( $array, 'sanitize_text_field' );
+
+		$array = array_filter( $array, function( $value ) {
```

Comment:
I'm getting an error here.

The error says that a string value was provided to array_filter.

I logged the value of $array, and it is being logged twice.

The first time it is an array. The second time it is a json_encoded string containing the same data.

So for some reason this is being run twice, and the second time, the data is already encoded.

Reply:
Did this only happen the first time you saved something? I'm wondering if it's related to this:
https://core.trac.wordpress.org/ticket/21989

It would probably be best to modify the data with this filter before it gets saved instead
https://developer.wordpress.org/reference/hooks/pre_update_option_option/

Reply:
I think you're right about that issue, but I think we can solve it with less code by using the WordPress core's ability to save an option as an array. I think it uses [serialize](https://www.php.net/manual/en/function.serialize.php) instead of JSON.

I tried it out. Here are the changes I did to use a serialized array instead JSON. I'll send you a patch file with the same changes for easy copying.

class-store.php
<img width="644" alt="Screenshot 2023-10-30 at 1 27 55 PM" src="https://github.com/thoughtis/cata-blocks/assets/226381/9c49b378-23dc-472d-ac7f-6fe781db8250">
<img width="678" alt="Screenshot 2023-10-30 at 1 28 02 PM" src="https://github.com/thoughtis/cata-blocks/assets/226381/fbff8fae-c208-4da0-b66c-2f610df67e72">
<img width="654" alt="Screenshot 2023-10-30 at 1 28 20 PM" src="https://github.com/thoughtis/cata-blocks/assets/226381/f620eadf-e786-49a4-95c3-3c362ef05674">


<img width="689" alt="Screenshot 2023-10-30 at 1 29 20 PM" src="https://github.com/thoughtis/cata-blocks/assets/226381/1b06f4e7-00a5-46d3-8d5c-29a7245a4698">